shared/test/test_dms.rb

```diff
@@ -0,0 +1,179 @@
+          tables_loading: 2,
+          full_load_progress_percent: 30
+        }
+      }
+    )
+
+    @task_completed_successfully = @task_starting.deep_merge(
+      {
+        status: 'stopped',
+        stop_reason: 'Stop Reason FULL_LOAD_ONLY_FINISHED',
+        replication_task_stats: {
+          elapsed_time_millis: 2000,
+          tables_queued: 0,
+          tables_loaded: 2,
+          full_load_progress_percent: 100
+        }
+      }
+    )
+
+    @task_completed_successfully_table_statistics = [
+      {
+        "schema_name": "my_favorite_schema",
+        "table_name": "my_favorite_table",
+        "inserts": 0,
+        "deletes": 0,
+        "updates": 0,
+        "ddls": 0,
+        "full_load_rows": 2_147_483_647,
+        "full_load_condtnl_chk_failed_rows": 0,
+        "full_load_error_rows": 0,
+        "last_update_time": Time.parse("2020-01-01T17:20:00.000-00:00"),
+        "table_state": "Table completed",
+        "validation_pending_records": 0,
+        "validation_failed_records": 0,
+        "validation_suspended_records": 0,
+        "validation_state": "Not enabled"
+      },
+      {
+        "schema_name": "my_favorite_schema",
+        "table_name": "my_bestie_table",
+        "inserts": 0,
+        "deletes": 0,
+        "updates": 0,
+        "ddls": 0,
+        "full_load_rows": 314159,
+        "full_load_condtnl_chk_failed_rows": 0,
+        "full_load_error_rows": 0,
+        "last_update_time": Time.parse("2020-01-01T17:21:00.000-00:00"),
+        "table_state": "Table completed",
+        "validation_pending_records": 0,
+        "validation_failed_records": 0,
+        "validation_suspended_records": 0,
+        "validation_state": "Not enabled"
+      }
+    ]
+
+    @task_completed_unsuccessfully = @task_starting.deep_merge(
+      {
+        status: 'stopped',
+        stop_reason: 'Stop Reason FULL_LOAD_ONLY_FINISHED',
+        replication_task_stats: {
+          elapsed_time_millis: 2000,
+          tables_loading: 1,
+          tables_loaded: 1,
+          full_load_progress_percent: 74
+        }
+      }
+    )
+
+    @task_completed_unsuccessfully_table_statistics = @task_completed_successfully_table_statistics.deep_dup
+    @task_completed_unsuccessfully_table_statistics[0].deep_merge({"table_state": "Table Loading"})
+
+    @start_task_response = {
+      "migration_type": "full-load",
+      "replication_instance_arn": "arn:aws:dms:us-east-1:1234567890:task:LKJIHGFEDCBA",
+      "replication_task_arn": @task_arn,
+      "replication_task_creation_date": Time.parse("2020-01-01T00:00:00.000-00:00"),
+      "replication_task_identifier": "my-favorite-task",
+      "replication_task_settings": "{\"escapedJSON\": true}",
+      "source_endpoint_arn": "arn:aws:dms:us-east-1:1234567890:task:ABCDEFGHIJKL",
+      "status": "starting",
+      "table_mappings": "{\"escapedJSON\": true}",
+      "target_endpoint_arn": "arn:aws:dms:us-east-1:1234567890:task:ZYXWVUTSRQPO"
+    }
+  end
```

Comment:
This set of fixtures should also be useful in any further work we do with DMS, which is helpful!